cheroot/ssl/builtin.py

```diff
@@ -309,6 +309,9 @@ def bind(self, sock):
 
     def wrap(self, sock):
         """Wrap and return the given socket, plus WSGI environ entries."""
+        if self._check_for_plain_http(sock):
+            raise errors.NoSSLError
```

Comment:
Can we start providing some context with the exception so that it can be logged?
```suggestion
            raise errors.NoSSLError('Expected HTTPS on the socket but got plain HTTP') from None
```